webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -65,6 +71,16 @@ export class MonacoService {
       }
     });
 
+    let self = this; // Monaco bug: editor.addAction only works on the left-hand side of the Diff viewer
+    document.addEventListener("keydown", function(e) { // Pure JS, Ctrl-S solution instead...
```

Comment:
if you add an event listener manually then you need to remove it manually. Is this a right place to attach an event listener?